cmd/finch/version_test.go

```diff
@@ -5,44 +5,177 @@ package main
+				//nolint: lll	// Version output format is larger than 500
+				command.EXPECT().Output().Return([]byte(`{"Client":{"Version":"v1.0.0","GitCommit":"c00780a1f5b905b09812722459c54936c9e070e6","GoVersion":"go1.19.2","Os":"linux","Arch":"arm64","Components":[{"Name":"buildctl","Version":"v0.10.5","Details":{"GitCommit":"bc26045116045516ff2427201abd299043eaf8f7"}}]},"Server":{"Components":[{"Name":"containerd","Version":"v1.6.8","Details":{"GitCommit":"9cd3357b7fd7218e4aec3eae239db1f68a5a6ec6"}},{"Name":"runc","Version":"1.1.4","Details":{"GitCommit":"v1.1.4-0-g5fd4c4d1"}}]}}`), nil)
+			},
+		},
+	}
 
-	f()
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
 
-	require.NoError(t, write.Close())
-	os.Stdout = temp
+			ctrl := gomock.NewController(t)
+			logger := mocks.NewLogger(ctrl)
+			lcc := mocks.NewLimaCmdCreator(ctrl)
+			var stdOut bytes.Buffer
+			tc.mockSvc(lcc, logger, ctrl)
 
-	var buffer bytes.Buffer
-	_, err := io.Copy(&buffer, read)
-	require.NoError(t, err)
-	return buffer.String()
+			assert.NoError(t, newVersionAction(lcc, logger, &stdOut).runAdapter(tc.cmd, tc.args))
+		})
+	}
 }
 
-func TestVersionCommand(t *testing.T) {
+func TestVersionAction_run(t *testing.T) {
 	t.Parallel()
 
-	cmd := newVersionCommand()
-	assert.Equal(t, cmd.Name(), "version")
-}
+	testCases := []struct {
+		name         string
+		wantErr      error
+		cmd          *cobra.Command
+		mockSvc      func(*mocks.LimaCmdCreator, *mocks.Logger, *gomock.Controller)
+		postRunCheck func(t *testing.T, stdout []byte)
+	}{
+		{
+			name:    "print finch version with Client and Server",
+			wantErr: nil,
+			cmd: &cobra.Command{
+				Use: "version",
+			},
+			mockSvc: func(lcc *mocks.LimaCmdCreator, logger *mocks.Logger, ctrl *gomock.Controller) {
+				getVMStatusC := mocks.NewCommand(ctrl)
+				lcc.EXPECT().CreateWithoutStdio("ls", "-f", "{{.Status}}", limaInstanceName).Return(getVMStatusC)
+				getVMStatusC.EXPECT().Output().Return([]byte("Running"), nil)
+				logger.EXPECT().Debugf("Status of virtual machine: %s", "Running")
+
+				command := mocks.NewCommand(ctrl)
+				lcc.EXPECT().CreateWithoutStdio("shell", limaInstanceName, "nerdctl", "version", "--format", "json").Return(command)
+				//nolint: lll	// Version output format is larger than 500
+				command.EXPECT().Output().Return([]byte(`{"Client":{"Version":"v1.0.0","GitCommit":"c00780a1f5b905b09812722459c54936c9e070e6","GoVersion":"go1.19.2","Os":"linux","Arch":"arm64","Components":[{"Name":"buildctl","Version":"v0.10.5","Details":{"GitCommit":"bc26045116045516ff2427201abd299043eaf8f7"}}]},"Server":{"Components":[{"Name":"containerd","Version":"v1.6.8","Details":{"GitCommit":"9cd3357b7fd7218e4aec3eae239db1f68a5a6ec6"}},{"Name":"runc","Version":"1.1.4","Details":{"GitCommit":"v1.1.4-0-g5fd4c4d1"}}]}}`), nil)
+			},
+			postRunCheck: func(t *testing.T, stdout []byte) {
+				lines := strings.SplitAfter(string(stdout), "\n")
+
```

Comment:
nit: prevent panic when there's a length mismatch

```suggestion
require.Len(t, lines, 18)
```